clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/DeleteAmazonSnapshotAtomicOperation.java

```diff
@@ -47,9 +57,16 @@ private static Task getTask() {
   @Override
   public Void operate(List priorOutputs) {
     getTask().updateStatus(BASE_PHASE, String.format("Initializing Delete Snapshot operation for %s", description));
-    amazonClientProvider
-      .getAmazonEC2(description.getCredentials(), description.getRegion())
-      .deleteSnapshot(new DeleteSnapshotRequest().withSnapshotId(description.getSnapshotId()));
+    try {
+      amazonClientProvider
+        .getAmazonEC2(description.getCredentials(), description.getRegion())
+        .deleteSnapshot(new DeleteSnapshotRequest().withSnapshotId(description.getSnapshotId()));
+    } catch (Exception e) {
+      registry.counter(deleteSnapshotTaskId.withTag("success", false)).increment();
+      log.error("Failed to delete snapshotId with exception: {}", e.getMessage());
```

Comment:
I'd prefer:

`log.error(format("Failed to delete snapshotId %s", description.getSnapshotId()), e);`

This way we actually get a stacktrace, and we get real context of _what_ snapshot failed. In the log's current form, we wouldn't have much actionable and may have a more challenging time sourcing the location / origin of this error.